core/embed/rust/src/ui/model_tr/component/flow_pages_helpers.rs

```diff
@@ -0,0 +1,177 @@
+//! Mostly extending TextLayout from ui/component/text/layout.rs
+//! (support for more Ops like alignments or arbitrary offsets)
+
+use crate::{
+    strutil::StringType,
+    ui::{
+        component::{
+            text::layout::{LayoutFit, LayoutSink},
+            TextLayout,
+        },
+        display::{Color, Font},
+        geometry::{Alignment, Offset, Point},
+    },
+};
+
+use heapless::Vec;
+
+/// To account for operations that are not made of characters
+/// but need to be accounted for somehow.
+/// Number of processed characters will be increased by this
+/// to account for the operation.
+const PROCESSED_CHARS_ONE: usize = 1;
+
+/// Container for text allowing for its displaying by chunks
+/// without the need to allocate a new String each time.
+#[derive(Clone)]
+pub struct ToDisplay<T> {
+    pub text: T,
+    pub length_from_end: usize,
+}
+
+impl<T> ToDisplay<T>
+where
+    T: StringType,
+{
+    pub fn new(text: T) -> Self {
+        Self {
+            text: text.clone(),
+            length_from_end: text.as_ref().len(),
+        }
+    }
+}
+
+/// Operations that can be done on the screen.
+#[derive(Clone)]
+pub enum Op<T>
+where
+    T: StringType,
+{
+    /// Render text with current color and font.
+    Text(ToDisplay<T>),
+    /// Set current text color.
+    Color(Color),
+    /// Set currently used font.
+    Font(Font),
+    /// Set currently used line alignment.
+    Alignment(Alignment),
+    /// Move the current cursor by specified Offset.
+    CursorOffset(Offset),
+    /// Force continuing on the next page.
+    NextPage,
+}
+
+impl TextLayout {
+    /// Perform some operations defined on `Op` for a list of those `Op`s
+    /// - e.g. changing the color, changing the font or rendering the text.
+    pub fn layout_ops_new<const M: usize, T: StringType>(
```

Comment:
What's the point of reimplementing `layout_ops` instead of extending the existing one?

Reply:
In recent commits I have improved upon this, now `Flow` uses `FormattedText` under the hood and gets it already completed as an argument, so does not need to delegate all those chaining-methods.

Also, it is currently possible to use `FormattedText` and its chaining-API instead of `Paragraphs`, so we could have two or more fonts at the same line